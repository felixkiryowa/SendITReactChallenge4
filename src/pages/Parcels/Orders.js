import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getUserParcelOrders } from '../../actions//parcelActions';
import CustomerOrdersComponent from '../../components/Parcels/CustomerOrders';

export class Orders extends Component {

    componentDidMount() {
        const { userOrders } = this.props;
        userOrders();
    }

    render() {
        return (
            <div>
                <CustomerOrdersComponent parcelOrders={ this.props.parcels } />
            </div>
        )
    }
}

Orders.propTypes = {
    userOrders: PropTypes.func.isRequired,
    history: PropTypes.shape({
        push: PropTypes.func.isRequired,
    }).isRequired,
};

const mapStateToProps = state => ({
    parcels: state.parcels.parcels,
});

export default connect(mapStateToProps, {
    userOrders: getUserParcelOrders,
})(Orders);
