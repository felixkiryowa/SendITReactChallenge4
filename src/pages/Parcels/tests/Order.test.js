import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Orders } from '../Orders';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
    parcelReducer: {
        parcels: [],
    },
};
const store = mockStore(initialState);


describe('Orders Page Component', () => {
    let mountApp;
    const props = {
        userOrders: jest.fn(),
        parcelOrders: [
            {
                "created_at": "Tue, 28 May 2019 08:04:50 GMT",
                "order_current_location": "Kasese, Uganda",
                "order_name": null,
                "order_status": "pending",
                "parcel_destination_address": "Kampala, Uganda",
                "parcel_order_id": 1,
                "parcel_pickup_address": "Kasese, Uganda",
                "parcel_weight": 4,
                "price": 200000,
                "receivers_contact": null,
                "receivers_names": null,
                "senders firstname": "Roy ",
                "senders lastname": "Wasibani",
                "senders phone contact": "0700162509",
                "senders_user_id": 5
            },
            {
                "created_at": "Tue, 28 May 2019 08:04:50 GMT",
                "order_current_location": "Lugazi, Mukono, Uganda",
                "order_name": null,
                "order_status": "pending",
                "parcel_destination_address": "Jinja, Uganda",
                "parcel_order_id": 2,
                "parcel_pickup_address": "Lugazi, Mukono, Uganda",
                "parcel_weight": 5,
                "price": 250000,
                "receivers_contact": null,
                "receivers_names": null,
                "senders firstname": "Roy ",
                "senders lastname": "Wasibani",
                "senders phone contact": "0700162509",
                "senders_user_id": 5
            }
        ],
    }
    it('Should render without crashing', () => {
        mountApp = shallow(
            <Router>
                <Provider store={store}>
                    <Orders {...props} />
                </Provider>
            </Router>,
        );
        expect(mountApp).toMatchSnapshot();
    });

    it('Testing componentDidMount Method Called', () => {
        const spy = jest.spyOn(Orders.prototype, 'componentDidMount');
        shallow(<Orders {...props} />);
        expect(spy).toHaveBeenCalledTimes(1);
    });

});
