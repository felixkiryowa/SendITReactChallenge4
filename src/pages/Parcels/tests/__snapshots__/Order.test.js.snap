// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Page Component Should render without crashing 1`] = `ShallowWrapper {}`;
